src/app/ecosystem-explorer/project-form/utils/fileUtils.ts

```diff
@@ -0,0 +1,59 @@
+import path from 'path'
+
+import type { EcosystemProject } from '@/types/ecosystemProjectType'
+
+import type { EcosystemProjectFormLogoData } from '../schema/EcosystemProjectFormSchema'
+
+export type FormattedLogo = Awaited<ReturnType<typeof formatLogo>>
+
+export async function formatLogo(logo: EcosystemProjectFormLogoData) {
+  const base64 = await convertToBase64(logo)
+  const format = getFileFormat(logo.name)
+
+  return { base64, format, name: logo.name }
+}
+
+export async function buildTemporaryLogoFile(image: EcosystemProject['image']) {
+  if (!image) {
+    return
+  }
+
+  const dummyFileName = 'temporaryLogo'
+  const response = await fetch(image.src)
+  const blob = await response.blob()
+
+  return new File([blob], dummyFileName, { type: blob.type })
+}
+
+function convertToBase64(file: File): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.readAsDataURL(file)
+    reader.onload = () => {
+      const rawResult = String(reader.result)
+      const [, cleanBase64String] = rawResult.split('base64,')
+      resolve(cleanBase64String)
+    }
+    reader.onerror = (error) => reject(error)
+  })
+}
+
+const ALLOWED_IMAGE_FORMATS = ['png', 'jpg', 'jpeg', 'svg', 'webp'] as const
```

Comment:
Should we use this in `ControlledFormFileInput`

Reply:
✅